app/experimenter/legacy-ui/rapid/components/experiments/utils.ts

```diff
@@ -0,0 +1,28 @@
+import { XSelectOption } from "experimenter-rapid/components/forms/XSelect";
+
+export const displaySelectOptionList = (
+  options: XSelectOption[],
+  values: string | string[],
+): string[] => {
+  let selectedValue: string | string[] = values;
+  if (!Array.isArray(values)) {
+    selectedValue = [values];
+  }
+
+  const selectedOption = options.reduce((filtered: string[], element) => {
+    if (selectedValue.includes(element.value)) {
+      filtered.push(element.label);
+    }
+
+    return filtered;
+  }, []);
```

Comment:
Is this just filtering an array? Maybe I'm misreading it. If it is, you could just do

```js
const selectedOption = options.filter(o => selectedValue.includes(o.value))
```

Reply:
Oh I see you're also returning the label string. In that case:

```js
const selectedOption = options.filter(o => selectedValue.includes(o.value)).map(o => o.label);
```

But that starts to get into the size of just using reduce, so I'm fine leaving it as you have it.